### PR TITLE
feat: automate container build with GitHub Actions

### LINKED ISSUE #50: Create the container image in GitHub Actions
`wws` is also provided as a container imager for convenience. However, this image is still generated manually. Since we plan to start using GitHub registry to publish them, we want to start using GitHub actions to publish it. These are the tags that should be created:

- A `preview` tag on every commin in `main`
- A `X.X.X` tag per new released version
- A `latest` tag pointing to the latest released version

### PR DESCRIPTION
**Pull request by Angelmmiguel**
Automate the creation of containers using GitHub Actions. This will push the images to the GitHub Package Registry. In addition to that, we are also trying to reduce the size of the produced images.

I created two different actions:

- Preview: runs on every commit in `main`
- Release: runs on every new vX.X.X tag

The code is duplicated, but I plan to refactor all actions and make some parts reusable in a separate PR.

Here you have the created package: https://github.com/vmware-labs/wasm-workers-server/pkgs/container/wws 

Closes #50 and #24

### CODE REVIEW

File: .github/workflows/container-preview.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+name: Container Build Preview
+
+on:
+  push:
+    branches: [ "main" ]
+  # TODO: uncomment for testing. This will run by default only when a new commit is push to main
+  # pull_request:
+  #   branches: [ "main" ]
```
Comment (ereslibre) [inline .github/workflows/container-preview.yml:10]:
```suggestion
```

I would remove this comment here, I don't think we want to try to build the preview for PR's

File: .github/workflows/container-release.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+# SPDX-License-Identifier: Apache-2.0
+name: Container Build Release
+
+on:
+  push:
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+  # TODO: uncomment for testing. This will run by default only when a new commit is push to main
+  # pull_request:
+  #   branches: [ "main" ]
```
Comment (ereslibre) [inline .github/workflows/container-release.yml:11]:
```suggestion
```

I would remove this comment. In order to try the workflow, I think is safer to fork on another place (like a personal account), and open the PR when everything is fine.

**Review by ereslibre — COMMENTED** [review]
Great! 👏, some comments.

File: .github/workflows/container-preview.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        include:
+          - arch: x86_64
+            cross: false
+            platform: unknown-linux-musl
+          - arch: aarch64
+            cross: true
+            platform: unknown-linux-musl
+    env:
+      # We use cross for arm builds
+      CARGO: cargo
```
[the remark this answers is not in the document]
Comment (Angelmmiguel) [reply .github/workflows/container-preview.yml]: Good point! I updated it 😄

File: .github/workflows/container-preview.yml (hunk 1 above)
Comment (Angelmmiguel) [reply .github/workflows/container-preview.yml]: Makes sense. I'm gonna remove it 👍

**Review by ereslibre — APPROVED** [review]
🚢 it! 👏

**Pull request merged**